src/main/scala/com/workday/warp/utils/AnnotationReader.scala

```diff
@@ -93,12 +79,10 @@ object AnnotationReader extends StackTraceFilter {
    * @param testId fully qualified name of the junit test method
    * @return max response time as a [[Duration]] for the test we are about to invoke
    */
-  // TODO maybe return an Option?
   def getRequiredMaxValue(testId: String): Duration = {
-    getWarpTestMethodAnnotation(classOf[Required], testId) match {
-      case None => Duration.ofMillis(-1)
-      case Some(required) => Duration.ofNanos(TimeUtils.toNanos(required.maxResponseTime, required.timeUnit))
-    }
+    this.getWarpTestMethodAnnotation(classOf[Required], testId)
+      .map(req => Duration.ofNanos(TimeUtils.toNanos(req.maxResponseTime, req.timeUnit)))
```

Comment:
👍

Reply:
i have found `match` is usually not the answer

Reply:
would fold be a better option here instead of a map().getOrElse?

Reply:
we will leave it up to preference. `fold` is more concise and elegant, but `getOrElse` is more straightforward

fold reads pretty well with a named parameter:

```scala
Option(3).fold(ifEmpty = 1)(_ + 1)
```